network-api/networkapi/wagtailpages/pagemodels/base.py

```diff
@@ -434,6 +436,60 @@ class ParticipateHighlights2(ParticipateHighlightsBase):
     )
 
 
+class HomepageTakeActionCards(WagtailOrderable, models.Model):
+    page = ParentalKey(
+        'wagtailpages.Homepage',
+        related_name='take_action_cards',
+    )
+    image = models.ForeignKey(
+        'wagtailimages.Image',
+        null=True,
+        blank=False,
+        on_delete=models.SET_NULL,
+    )
+    text = models.CharField(max_length=255)
+    internal_link = models.ForeignKey(
```

Comment:
This works, but also feels like we should check with the wagtail folks whether this is a solved problem: two different fields for internal vs. external links should really be a single field that lets you specify whether you want a `Page` or a bare URL on a single input field.

Ideally, let's put a source comment here that points to either an issue on our repo, or an issue on the wagtail issue tracker, that might inform us of a prebuilt solution that we can change this to before the eventual merge into master.

It would be very curious if we were the first folks to ever need this =)

Reply:
There's an issue open for this from 2014. This is one of those major pains we've been feeling for a long time. https://github.com/wagtail/wagtail/issues/381 

As far as I know, there aren't any packages that do (or attempt to do) this, though I'd love if I was wrong about that. 

I'll likely write an RFC for Wagtail to have this in core.